include/envoy/secret/secret_manager.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <string>
+
+#include "envoy/secret/secret.h"
+
+namespace Envoy {
+namespace Secret {
+
+/**
+ * A manager for all static secrets.
+ */
+class SecretManager {
+public:
+  virtual ~SecretManager() {}
+
+  /**
+   * @param secret Updated Secret.
+   */
+
+  virtual void addOrUpdateStaticSecret(const SecretSharedPtr secret) PURE;
```

Comment:
nit: `const` doesn't do anything here, just remove. Better would probably be to make this a `SecretConstSharedPtr` and have const inside the pointer definition.

Reply:
Removed